python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -504,6 +505,24 @@ def makeMinimalSchema(filterNameList, *, addCentroid=False,
                 type="Flag",
                 doc="Set if parallax or parallax error is bad",
             )
+            if fullCoordinateCovariance:
+                if not (addProperMotion and addParallax):
+                    warnings.warn("Adding proper motion and parallax covariances without proper motion and "
+                                  "parallax parameters and errors.")
+                fields = ["coord_ra", "coord_dec", "pm_ra", "pm_dec", "parallax"]
+                units = ["rad", "rad", "rad/year", "rad/year", "rad"]
+                for i in range(5):
+                    i_field = fields[i]
+                    i_unit = units[i]
```

Comment:
I don't understand how this nested for-loop is an improvement over using the CovarianceMatrixXKey?

Reply:
The difference is that the resulting names match the existing schema much more closely, so we don't have to add hundreds of lines elsewhere.